subprojects/docs/src/samples/userguide/files/inputFiles/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,38 @@
+// tag::set-input-files[]
+task<JavaCompile>("compile") {
+    // Use a File object to specify the source directory
+    source = fileTree(file("src/main/java"))
+
+    // Use a String path to specify the source directory
+    source = fileTree("src/main/java")
+
+    // Use a collection to specify multiple source directories
+    source = fileTree(listOf("src/main/java", "../shared/java"))
+
+    // Use a FileCollection (or FileTree in this case) to specify the source files
+    source = fileTree("src/main/java").matching { include("org/gradle/api/**") }
+
+    // Using a closure to specify the source files.
+    // TODO how to do that?
+    /*
+    source = {
+        // Use the contents of each zip file in the src dir
+        file("src").listFiles().filter { it.name.endsWith(".zip") }.map { zipTree(it) }
+    }
+    */
```

Comment:
Using a `Provider<T>` instead of a Closure works here:
```kotlin
    // Use the contents of each zip file in the src dir
    setSource(provider {
        file("src").listFiles().filter { it.name.endsWith(".zip") }.map { zipTree(it) }
    })
```